cosmos/x/evm/plugins/txpool/mempool/mempool_writer.go

```diff
@@ -48,12 +48,13 @@ func (etp *EthTxPool) Insert(ctx context.Context, tx sdk.Tx) error {
 
 		// Reject txs with a nonce lower than the nonce reported by the statedb.
 		if sdbNonce := etp.nr.GetNonce(sender); sdbNonce > nonce {
-			return errors.New("nonce too low")
+			return errorslib.Wrap(etp.PriorityNonceMempool.Remove(tx), "nonce too low")
```

Comment:
have to remove from the underlying mempool bc it has already been inserted at this point